dasharo-stability/capsule-update.robot

```diff
@@ -240,3 +246,13 @@ Enter Capsule Testing Folder
     END
     ${out}=    Execute UEFI Shell Command    cd capsule_testing
     Should Not Contain    ${out}    is not a directory.
+
+Turn Off Active ME
+    IF    '${CONFIG}' != 'qemu'
```

Comment:
Changed in: 9918a92ee4a56e1b708d2015457651ac274bc7c7

Reply:
> I wonder if we have any ME platforms without neutering options, capsule support for such platforms would be complicated.

This might take making on-disk capsules work.

Reply:
I just tried it in QEMU and it didn't work because `${DASHARO_INTEL_ME_MENU_SUPPORT}` is true there but the menu is absent.  This might be due to the switch to Q35.  `platform-configs/qemu.robot` probably needs to be updated or this `IF` should exclude QEMU explicitly if that config can still be used with an older Dasharo QEMU version.  I don't  really know whether tests have made the transition but judging by the incorrect config they didn't and the latter path should be taken.

Reply:
Ah, because we have two coreboot config files for Q35, one with every menu visible, and the other one with only reasonable options. Tests were made with full menu in mind, as this is (was?) run as part of tests CI. I don't have any suggestion about how to get out of this problem in a way that won't create different ones. Perhaps renaming current `platform-configs/qemu.robot` to something like `platform-configs/qemu-selftests.robot` and modifying CI accordingly would be the best way forward, then we could add a config for "normal" Q35.

> This might take making on-disk capsules work.

IIUC this is how fwupd passes capsules, so we may want to do this anyway, but it wouldn't be part of the current project.

Reply:
> Tests were made with full menu in mind, as this is (was?) run as part of tests CI.

Right, but capsule updates don't work with "full" version of Q35 due to ME (can be fixed by adding new Kconfig options tough).

> > This might take making on-disk capsules work.
> 
> IIUC this is how fwupd passes capsules, so we may want to do this anyway, but it wouldn't be part of the current project.

Those aren't on-disk capsules as defined by UEFI.  Rather it's their alternative implementation: fwupd-efi simply calls `UpdateCapsule` ([here](https://github.com/fwupd/fwupd-efi/blob/3ea8a14bd7df23a9924eaf6ad2544b90747ef573/efi/fwupdate.c#L480)) to work around OSes that don't expose the call.

Reply:
> Perhaps renaming current `platform-configs/qemu.robot` to something like `platform-configs/qemu-selftests.robot` and modifying CI accordingly would be the best way forward, then we could add a config for "normal" Q35.

Added QEMU stuff (self-tests and these tests work in QEMU):

https://github.com/Dasharo/open-source-firmware-validation/pull/457/commits/ffe6d8737e1b6b10db816602aa1910fa9c2d28ec
https://github.com/Dasharo/open-source-firmware-validation/pull/457/commits/fc2826aa290bc12943f04f6b928b39aaf4795d5f
https://github.com/Dasharo/open-source-firmware-validation/pull/457/commits/b5264146bb58b7ef55a3f2eea972a75dacd240bf